sdk/package/src/services/accounts-service.ts

```diff
@@ -0,0 +1,217 @@
+     *
+     * @param regRequest Body of the scanned QR code
+     * @param fcmToken Firebase push authentication token
+     */
+    public async addAccount(regRequest: any, fcmToken: string): Promise<any> {
+
+        const discoveryData = this.processDiscoveryData(regRequest);
+
+        const keypair: any = CryptoUtil.generateKeypair();
+        const signatureString = regRequest.chg + "." + fcmToken;
+        let signedChallenge: string;
+
+        try {
+            signedChallenge = CryptoUtil.signChallenge(
+                keypair.prvKey,
+                signatureString
+            );
+        } catch (err) {
+
+            return JSON.stringify({data: null, res: "FAILED"});
+        }
+
+        const modPubKey: string = keypair.pubKey
+            .replace("-----BEGIN PUBLIC KEY-----", "")
+            .replace("-----END PUBLIC KEY-----", "")
+            .replace(/(\r\n|\n|\r)/gm, "");
+
+        const request: RegistrationRequestInterface = {
+            deviceId: discoveryData.deviceId,
+            publicKey: modPubKey,
+            pushID: fcmToken,
+            signature: signedChallenge
+        };
+
+        request.deviceName = DeviceInfoUtil.getDeviceName();
+        request.model = DeviceInfoUtil.getDeviceModel();
+
+        const regRequestBody: any = {
+            deviceId: request.deviceId,
+            model: request.model,
+            name: request.deviceName,
+            publicKey: request.publicKey,
+            pushId: request.pushID,
+            signature: request.signature
+        };
+
+        const requestMethod = "POST";
+
+        const headers = {
+            Accept: "application/json",
+            "Content-Type": "application/json"
+        };
+
+        const newRequest: RequestSenderUtil = new RequestSenderUtil();
+        const registrationUrl =
+            discoveryData.host +
+            discoveryData.basePath +
+            discoveryData.registrationEndpoint;
+
+        return newRequest
+            .sendRequest(
+                registrationUrl,
+                requestMethod,
+                headers,
+                JSON.stringify(regRequestBody)
+            )
+            .then((result) => {
+                let account: AccountsInterface;
+                if (result.status === 201) {
+                    account = {
+                        deviceID: request.deviceId,
+                        username: discoveryData.username,
+                        lastName: discoveryData.lastName,
+                        tenantDomain: discoveryData.tenantDomain,
+                        host: discoveryData.host,
+                        basePath: discoveryData.basePath,
+                        authenticationEndpoint: discoveryData.authenticationEndpoint,
+                        removeDeviceEndpoint: discoveryData.removeDeviceEndpoint,
+                        privateKey: keypair.prvKey
+                    };
+
+                    return JSON.stringify({res: "OK", data: account});
```

Comment:
Can't we return the object directly without stringifying?

Reply:
Tracked by #105